satpy/tests/test_config.py

```diff
@@ -480,6 +480,22 @@ def test_bad_str_config_path(self):
         with satpy.config.set(config_path='/single/string/paths/are/bad'):
             pytest.raises(ValueError, satpy._config.get_config_path_safe)
 
+    def test_defaults(self):
+        """Check default config."""
+        import satpy
+        keys_exp = {
+            'tmp_dir',
+            'cache_dir',
+            'cache_lonlats',
+            'cache_sensor_angles',
+            'config_path',
+            'data_dir',
+            'demo_data_dir',
+            'download_aux',
+            'sensor_angles_position_preference',
+        }
+        assert set(satpy.config.to_dict().keys()) == keys_exp
```

Comment:
I'm not sure I like tests like these. We've had a couple tests like these in the past with stuff like DataArray.attrs checks and I always get annoyed when I change one small thing and have to update a ton of tests that weren't actually doing anything very useful. The overall contents of the config object doesn't really matter as long as it is usable. Technically users can add their own keys just by defining `SATPY_*` environment variables. We only define defaults in `_config.py`. This test will fail if anything else is defined.

How about a test that gets the tmp dir value and makes sure it is a directory? Maybe even a writable directory?

Reply:
Good idea, I'll try that!